src/SphericalCoordinates_TEST.cc

```diff
@@ -383,3 +432,210 @@ TEST(SphericalCoordinatesTest, AssignmentOp)
   math::SphericalCoordinates sc2 = sc1;
   EXPECT_EQ(sc1, sc2);
 }
+
+//////////////////////////////////////////////////
+TEST(SphericalCoordinatesTest, NoHeading)
+{
+  // Default heading
+  auto st = math::SphericalCoordinates::EARTH_WGS84;
+  math::Angle lat(IGN_DTOR(-22.9));
+  math::Angle lon(IGN_DTOR(-43.2));
+  math::Angle heading(0.0);
+  double elev = 0;
+  math::SphericalCoordinates sc(st, lat, lon, elev, heading);
+
+  // Origin matches input
+  auto latLonAlt = sc.SphericalFromLocalPosition({0, 0, 0});
+  EXPECT_DOUBLE_EQ(lat.Degree(), latLonAlt.X());
+  EXPECT_DOUBLE_EQ(lon.Degree(), latLonAlt.Y());
+  EXPECT_DOUBLE_EQ(elev, latLonAlt.Z());
+
+  auto xyzOrigin = sc.LocalFromSphericalPosition(latLonAlt);
+  EXPECT_EQ(math::Vector3d::Zero, xyzOrigin);
+
+  // Check how different lat/lon affect the local position
+
+  // Increase latitude == go North == go +Y
+  {
+    auto xyz = sc.LocalFromSphericalPosition(
+        {lat.Degree() + 1.0, lon.Degree(), elev});
+    EXPECT_NEAR(xyzOrigin.X(), xyz.X(), 1e-6);
+    EXPECT_LT(xyzOrigin.Y(), xyz.Y());
+  }
+
+  // Decrease latitude == go South == go -Y
+  {
+    auto xyz = sc.LocalFromSphericalPosition(
+        {lat.Degree() - 1.0, lon.Degree(), elev});
+    EXPECT_NEAR(xyzOrigin.X(), xyz.X(), 1e-6);
+    EXPECT_GT(xyzOrigin.Y(), xyz.Y());
+  }
+
+  // Increase longitude == go East == go +X
+  // Also move a bit -Y because this is the Southern Hemisphere
+  {
+    auto xyz = sc.LocalFromSphericalPosition(
+        {lat.Degree(), lon.Degree() + 1.0, elev});
+    EXPECT_LT(xyzOrigin.X(), xyz.X());
+    EXPECT_GT(xyzOrigin.Y(), xyz.Y());
+  }
+
+  // Decrease longitude == go West == go -X
+  // Also move a bit -Y because this is the Southern Hemisphere
+  {
+    auto xyz = sc.LocalFromSphericalPosition(
+        {lat.Degree(), lon.Degree() - 1.0, elev});
+    EXPECT_GT(xyzOrigin.X(), xyz.X());
+    EXPECT_GT(xyzOrigin.Y(), xyz.Y());
+  }
+
```

Comment:
Tests for increasing/decreasing altitude?  Should be consistent because of right hand rule, but a reasonable sanity check.